### ISSUE TITLE
TiledMapPacker doesn't process animated tiles with strip unused

### ISSUE THREAD

**Issue by oddlydrawn** [libgdx/libgdx#3080]
I am using TiledMapPacker from the latest nightly to process multiple maps using a single, very large, shared tileset that consists of several different tilesets. TiledMapPacker can only process non animated tiles, completely ignoring the frames used in an animation.

##### Expected output:

All tiles added into texture atlas, including animated tiles when using the --strip-unused option.

##### Actual output:

Animated tiles are skipped when using the --strip-unused option.

##### Things I've tried include: 

not stripping unused tiles (impractical if I hope to remain under 1024x1024  or 2048x2048 sized tileset -- it's really big), and using a layer that isn't rendered to add each individual tile used in every animation e.g. a collision tile layer (impractical: tedious, waste of time, makes for unsightly layers).

##### Commands and resources used: 

    java -cp gdx.jar:gdx-natives.jar:gdx-backend-lwjgl.jar:gdx-backend-lwjgl-natives.jar:gdx-tiled-preprocessor.jar:extensions/gdx-tools/gdx-tools.jar com.badlogic.gdx.tiledmappacker.TiledMapPacker ../in/ ../out/ --strip-unused

Where the directory "../in/" uses gdx-tests-android's data/maps/tiled/tiles.png and a Tiled map set up for animation. [(files in "../in/")](https://github.com/oddlydrawn/libgdx-contrib/tree/master/tiledMapPacker/issueNonAnim/in) and [(files processed into "../out/")](https://github.com/oddlydrawn/libgdx-contrib/tree/master/tiledMapPacker/issueNonAnim/out) 

Note: Actually copy pasta everything into Eclipse isn't wise or advisable since you can just look at [out/map](https://github.com/oddlydrawn/libgdx-contrib/blob/master/tiledMapPacker/issueNonAnim/out/tiledPacker.tmx) and the atlas [out/tileset/packed.atlas](https://github.com/oddlydrawn/libgdx-contrib/blob/master/tiledMapPacker/issueNonAnim/out/tileset/packed.atlas) or the [tileset](https://github.com/oddlydrawn/libgdx-contrib/blob/master/tiledMapPacker/issueNonAnim/out/tileset/packed.png) to verify that the animated frames weren't processed.

##### To reproduce:

I copied the [/out/*](https://github.com/oddlydrawn/libgdx-contrib/tree/master/tiledMapPacker/issueNonAnim/out) processed atlas/map into gdx-tests-android: data/maps/tiled/ and used [Barebones.java](https://github.com/oddlydrawn/libgdx-contrib/blob/master/tiledMapPacker/issueNonAnim/Barebones.java) in gdx-tests-lwjgl. Which gives you the following exception:

#####  Exception when you try to render the processed, stripped tiled map:

    Exception in thread "LWJGL Application" java.lang.NullPointerException
	at com.badlogic.gdx.maps.tiled.tiles.AnimatedTiledMapTile.getTextureRegion(AnimatedTiledMapTile.java:84)
	at com.badlogic.gdx.maps.tiled.renderers.OrthogonalTiledMapRenderer.renderTileLayer(OrthogonalTiledMapRenderer.java:102)
	at com.badlogic.gdx.maps.tiled.renderers.BatchTiledMapRenderer.render(BatchTiledMapRenderer.java:138)
	at com.badlogic.gdx.tests.lwjgl.Barebones.render(Barebones.java:37)
	at com.badlogic.gdx.backends.lwjgl.LwjglApplication.mainLoop(LwjglApplication.java:215)
	at com.badlogic.gdx.backends.lwjgl.LwjglApplication$1.run(LwjglApplication.java:120)

##### Versions of stuffs:

Tiled Map Editor: 0.11.0

libGDX: latest nightly/git

Eclipse: Luna Service Release 2 (4.4.2). 

OS: Ubuntu 14.10 Utopic x86_64

##### TL;DR

I have code that would add animated tile processing for libGDX's TiledMapPacker when you feel the need to --strip-unused.

**Comment by MobiDevelop** [comment]
Honestly, I'd be inclined to remove the TiledMapPacker since it has pretty limited usefulness.

**Comment by oddlydrawn** [comment]
I've seen that talk in the forums and previous issues. Once my output texture atlas gets too big, I was thinking of looking into how to process a directory full of maps the way TexturePacker packs directories.. probably by liberating some of that heavily used/vetted code inside TexturePacker. I think? Maybe?

**Comment by Darkyenus** [comment]
It could be added to that development extension mentioned somewhere here

**Comment by oddlydrawn** [comment]
Here's the [branch diff](https://github.com/libgdx/libgdx/compare/master...oddlydrawn:tiledMapPackerAnimTiles) that would fix this issue.

**Comment by MobiDevelop** [comment]
If people still find the tool useful and are willing to fix or work around its shortcomings, I don't mind keeping it around.

**Comment by oddlydrawn** [comment]
I think more people would find the tool useful if there was a wiki explaining usage and caveats.

Is it alright to submit a pr of that diff, without the history? I'm unsure since CONTRIBUTING.md has "The issue tracker is the preferred channel for bug reports, features requests and submitting pull requests..."

**Comment by MobiDevelop** [comment]
Yeah, go ahead and submit a PR
